kernel/virtio/virtio_net.c

```diff
@@ -182,10 +183,15 @@ void virtio_config_network(struct pci_config_info *pci)
     virtq_init_rings(pci->base, &recvq, VIRTQ_RECV);
     virtq_init_rings(pci->base, &xmitq, VIRTQ_XMIT);
 
-    recvq.bufs = calloc(recvq.num, sizeof (struct io_buffer));
+    pgs = (recvq.num * sizeof (struct io_buffer) + PAGE_SIZE - 1) / PAGE_SIZE;
```

Comment:
As in `virtio_config_block`, use bit shifting in the same calculations in this function.